### PR TITLE
✨ Allow organizers to comment on a release

### PR DESCRIPTION
**Pull request by rahmonov**
This PR closes/references issue #772. It does so by adding a comment field to the schedule model.

## Screenshots (if appropriate):

image

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My change is listed in the CHANGELOG.rst if appropriate.

### CODE REVIEW

**Comment by rahmonov** [comment]
@rixx could you please take a look? If this looks good, I will fix the unit tests. Quite a bit of them are broken because of the changes to release_schedule and freeze methods.

File: src/pretalx/agenda/templates/agenda/changelog_block.html [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {% load i18n %}
+{% if schedule.comment %}
+  <p>{{ schedule.comment }}</p>
```
Comment (rixx) [inline src/pretalx/agenda/templates/agenda/changelog_block.html]: I'm thinking we should allow Markdown in the comment field, and then render it with our ``|rich_text`` template tag here.

File: src/pretalx/event/models/event.py [hunk 2]
```diff
@@ -781,16 +781,17 @@ def get_date_range_display(self) -> str:
         """
         return daterange(self.date_from, self.date_to)
 
-    def release_schedule(self, name: str, user=None, notify_speakers: bool = False):
+    def release_schedule(self, name: str, comment: str, user=None, notify_speakers: bool = False):
```
Comment (rixx) [inline src/pretalx/event/models/event.py]: Please make the comment optional, defaulting to ``None``, and move it to the end of the method signature – that way less tests should break, too.

File: src/pretalx/schedule/models/schedule.py [hunk 3]
```diff
@@ -48,11 +56,12 @@ class urls(EventUrls):
         public = "{self.event.urls.schedule}v/{self.url_version}/"
 
     @transaction.atomic
-    def freeze(self, name: str, user=None, notify_speakers: bool = True):
+    def freeze(self, name: str, comment: str, user=None, notify_speakers: bool = True):
```
Comment (rixx) [inline src/pretalx/schedule/models/schedule.py]: Please default to ``None`` here, as well, and put the comment to the end of the list of arguments. (I'm pretty sure we never call this method without explicit arguments in pretalx, but plugins may differ.)

File: src/pretalx/schedule/models/schedule.py [hunk 4]
```diff
@@ -37,6 +38,13 @@ class Schedule(LogMixin, models.Model):
         max_length=190, null=True, blank=True, verbose_name=_("version")
     )
     published = models.DateTimeField(null=True, blank=True)
+    comment = I18nTextField(
+        help_text=_(
+            "This text will be shown in the public changelog and the RSS feed."
```
Comment (rixx) [inline src/pretalx/schedule/models/schedule.py]: Could you please add ``phrases.use_markdown`` here? It's used in ``pretalx/submission/models/question.py`` in a similar place, for example.

File: src/pretalx/orga/views/schedule.py [hunk 5]
```diff
@@ -141,6 +142,7 @@ def form_invalid(self, form):
     def form_valid(self, form):
         self.request.event.release_schedule(
             form.cleaned_data["version"],
+            form.cleaned_data["comment"],
```
Comment (rixx) [inline src/pretalx/orga/views/schedule.py]: Just a reminder, as per my other comments I'd prefer keyword arguments for this method, so please adjust the call here, too.

File: src/pretalx/orga/forms/schedule.py [hunk 6]
```diff
@@ -1,10 +1,11 @@
 from django.forms import BooleanField, ModelForm, ValidationError
 from django.utils.translation import gettext_lazy as _
+from i18nfield.forms import I18nModelForm
 
 from pretalx.schedule.models import Schedule
 
 
-class ScheduleReleaseForm(ModelForm):
+class ScheduleReleaseForm(ModelForm, I18nModelForm):
```
Comment (rixx) [inline src/pretalx/orga/forms/schedule.py]: ``I18nModelForm`` subclasses ``ModelForm``, so we only need ``I18nModelForm`` here.

**Pull request merged**